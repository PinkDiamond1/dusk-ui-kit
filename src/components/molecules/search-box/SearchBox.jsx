/* @flow */
import * as React from 'react'
import BootstrapForm from 'react-bootstrap/Form'
import BootstrapInputGroup from 'react-bootstrap/InputGroup'
import BootstrapFormControl from 'react-bootstrap/FormControl'
import BootstrapButton from 'react-bootstrap/Button'

import MagnifierSvg from './SearchBox.svg'

type Props = {
  className: string,
  placeholder: string,
  onSubmit: Function,
  onChange: Function,
}

const SearchBox = (props: Props): React.Element<*> => {
  const { className, placeholder, onSubmit, onChange } = props
  const updatedClassName = `__duk-search-box ${className}`
  return (
    <BootstrapForm inline onSubmit={onSubmit} className={updatedClassName}>
      <BootstrapInputGroup>
        <BootstrapFormControl
          onChange={onChange}
          size="lg"
          type="search"
          placeholder={placeholder}
        />
        <BootstrapInputGroup.Append>
          <BootstrapButton type="submit" variant="primary" size="lg">
            <MagnifierSvg width="24" height="24" />
          </BootstrapButton>
        </BootstrapInputGroup.Append>
      </BootstrapInputGroup>
    </BootstrapForm>
  )
}

SearchBox.defaultProps = {
  className: '',
  placeholder: 'Search',
  onSubmit: () => {},
  onChange: () => {},
}

export default SearchBox
