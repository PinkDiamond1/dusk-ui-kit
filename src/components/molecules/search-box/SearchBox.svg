<svg xmlns="http://www.w3.org/2000/svg" width="27.828" height="27.828" viewBox="0 0 27.828 27.828"><defs><style>.a,.b{fill:none;stroke:#fff;stroke-miterlimit:10;}.a{stroke-width:2px;}.b{stroke-linecap:round;stroke-width:4px;}</style></defs><g transform="translate(-1494.454 -152.51)"><rect class="a" width="16.873" height="16.873" rx="8.436" transform="translate(1495.454 153.51)"/><line class="a" x2="8.226" y2="8.226" transform="translate(1510.167 168.222)"/><line class="b" x2="6.04" y2="6.04" transform="translate(1513.414 171.47)"/></g></svg>