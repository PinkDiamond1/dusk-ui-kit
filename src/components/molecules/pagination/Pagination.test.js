import React from 'react'
import renderer from 'react-test-renderer'

import SearchBox from './Pagination'

it('test to see if Pagination renders correctly', () => {
  const tree = renderer.create(<SearchBox />).toJSON()
  expect(tree).toMatchSnapshot()
})
