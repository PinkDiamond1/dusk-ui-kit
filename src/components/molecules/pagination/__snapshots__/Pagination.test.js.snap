// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if Pagination renders correctly 1`] = `
<ul
  className="__duk-pagination  pagination"
/>
`;
